pkg/kubernetes/cluster.go

```diff
@@ -196,13 +197,26 @@ func setClusterCNI(config map[string]any) {
 	config[cniKey] = cniDefaultValue
 }
 
-func setClusterAPIAddress(config map[string]any, apiAddress string, port int) {
-	if apiAddress == "" {
-		zap.S().Warn("Attempted to set an empty cluster API address")
+func setClusterAPIAddress(config map[string]any, apiAddress string, port uint16) {
```

Comment:
We must ensure that this function is successful, because if we don't, then a cluster will not be formed at all. The checks below should be part of the [validation logic](https://github.com/suse-edge/edge-image-builder/blob/main/pkg/image/validation/kubernetes.go) instead.